test/debug_info/deferred_len_in_module.f90

```diff
@@ -0,0 +1,13 @@
+!RUN: %flang -g -S -emit-llvm %s -o - | FileCheck %s
```

Comment:
IMO all of the newly added test cases should begin with this clause:
```
!
! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
! See https://llvm.org/LICENSE.txt for license information.
! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
!
```

Reply:
Added the clause.